### PR TITLE
mgr/dashboard: add validator for size field in the forms 

### PR DESCRIPTION
**Pull request by nizamial09**
**BEFORE**

https://github.com/ceph/ceph/assets/71764184/35ff4d95-64a6-4b69-9a2f-886c7543110f

**AFTER**

https://github.com/ceph/ceph/assets/71764184/266cd612-0289-4322-9d07-1ce5f0370cfc

Fixes: https://tracker.ceph.com/issues/62655

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Review by Pegonzal — CHANGES_REQUESTED** [review]
Looks like its failing for some cases where the input field is not named `size`.

Also some test are failing due to the changes, will need to adapt those.

**Comment by nizamial09** [comment]
thanks @Pegonzal, updated the PR

**Review by Pegonzal — APPROVED** [review]

**Comment by avanthakkar** [comment]
jenkins test make check

**Comment by avanthakkar** [comment]
jenkins test dashboard cephadm

File: src/pybind/mgr/dashboard/frontend/src/app/shared/directives/dimless-binary.directive.ts [hunk 1]
```diff
@@ -86,6 +86,12 @@ export class DimlessBinaryDirective implements OnInit {
   setValue(value: string) {
     if (/^[\d.]+$/.test(value)) {
       value += this.defaultUnit || 'm';
+    } else {
+      if (value) {
+        this.control.control.setValue(value);
+        this.control.control.addValidators(Validators.pattern(/^[a-zA-Z\d. ]+$/));
```
Comment (avanthakkar) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/directives/dimless-binary.directive.ts:92]: Don't we already have a validator for checking a valid number?

Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/directives/dimless-binary.directive.ts:92]: but this checks number and alphabets along with a space.

**Review by aaSharma14 — APPROVED** [review]

**Pull request merged**